MBBSEmu/Server/Socket/SocketServer.cs

```diff
@@ -23,15 +25,17 @@ public class SocketServer : ISocketServer
         private System.Net.Sockets.Socket _listenerSocket;
         private EnumSessionType _sessionType;
         private string _moduleIdentifier;
+        private readonly PointerDictionary<SessionBase> _channelDictionary;
 
-        public SocketServer(ILogger logger, IMbbsHost host, AppSettings configuration)
+        public SocketServer(ILogger logger, IMbbsHost host, AppSettings configuration, PointerDictionary<SessionBase> channelDictionary)
         {
             _logger = logger;
             _host = host;
             _configuration = configuration;
+            _channelDictionary = channelDictionary;
         }
 
-        public void Start(EnumSessionType sessionType, int port, string moduleIdentifier = null)
+        public void Start(EnumSessionType sessionType, int port, PointerDictionary<SessionBase> channelDictionary, string moduleIdentifier = null)
```

Comment:
We don't need to pass in channel dictionary on start since it's included in the constructor